### ISSUE TITLE
Make ClusterFactory more resliliant

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#10274]
This error is preventing the cache from wiring and binding to port 7800.  This is due to the heartbeat code getting corrupt data regarding the cluster from the DB and  then the system not seeing the proper heartbeat files in the /assets file share.  

It also seems that none of this should happen if `CLUSTER_AUTOWIRE` is set to false.

```
[04/12/16 10:37:00:647 CST]  INFO startup.StartupTasksExecutor: Not running: Task00001LoadSchema
[04/12/16 10:37:00:650 CST]  INFO startup.StartupTasksExecutor: Running: Task00003CreateSystemRoles
[04/12/16 10:37:01:127 CST]  INFO servlet.DispatcherServlet: FrameworkServlet 'servlet_1': initialization completed in 1534 ms
[04/12/16 10:37:01:231 CST]  INFO startup.StartupTasksExecutor: Not running: Task00004LoadStarter
[04/12/16 10:37:01:233 CST]  INFO startup.StartupTasksExecutor: Not running: Task00005LoadFixassets
[04/12/16 10:37:01:235 CST]  INFO startup.StartupTasksExecutor: Running: Task00006CreateSystemLayout
[04/12/16 10:37:01:246 CST]  INFO guava.GuavaCache: ***	 Building Cache : dotcmslayoutcache, size:10000,Concurrency:350
[04/12/16 10:37:01:292 CST]  INFO startup.StartupTasksExecutor: Not running: Task00007RemoveSitesearchQuartzJob
[04/12/16 10:37:01:301 CST]  INFO startup.StartupTasksExecutor: Running: Task00009ClusterInitialize
[04/12/16 10:37:01:359 CST] ERROR cluster.ClusterFactory: Error trying to wire node to the Cluster
java.lang.Exception: Assets folder of this node needs to point to /Assets of the master node to join the Cluster
	at com.dotcms.enterprise.cluster.ClusterFactory.addNodeToCluster(SourceFile:230) ~[ee-g0951ce8.jar:?]
	at com.dotcms.enterprise.cluster.ClusterFactory.addNodeToCluster(SourceFile:198) ~[ee-g0951ce8.jar:?]
	at com.dotcms.enterprise.cluster.ClusterFactory.initialize(SourceFile:112) [ee-g0951ce8.jar:?]
	at com.dotmarketing.startup.runalways.Task00009ClusterInitialize.executeUpgrade(Task00009ClusterInitialize.java:45) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
	at com.dotmarketing.startup.StartupTasksExecutor.executeUpgrades(StartupTasksExecutor.java:209) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
	at com.liferay.portal.servlet.MainServlet.init(MainServlet.java:116) [dotcms_3.5.1_3.5-8-g8506266.jar:?]
	at org.apache.catalina.core.StandardWrapper.initServlet(StandardWrapper.java:1231) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardWrapper.loadServlet(StandardWrapper.java:1144) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardWrapper.load(StandardWrapper.java:1031) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardContext.loadOnStartup(StandardContext.java:4901) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5188) [catalina.jar:8.0.18]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:725) [catalina.jar:8.0.18]
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:701) [catalina.jar:8.0.18]
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714) [catalina.jar:8.0.18]
```

**Comment by nollymar** [comment]
PR: core (https://github.com/dotCMS/core/pull/10291) and enterprise (https://github.com/dotCMS/enterprise-2.x/pull/397)

**Comment by bryanboza** [comment]
Fixed, tested on master-4.0 // MySQL // Chrome
